src/utils/array-utils.tsx

```diff
@@ -0,0 +1,25 @@
+/**
+ * Returns a new array with shuffled cards
+ * @param cards card piles
+ */
+export function shuffleArray<T = unknown>(cards: readonly T[]): T[] {
+  const newPile = cards.slice();
+  let currentIndex = newPile.length;
+  let temporaryValue;
+  let randomIndex;
+
+  // While there remain elements to shuffle...
+  while (currentIndex !== 0) {
+    // Pick a remaining element...
+    randomIndex = Math.floor(Math.random() * currentIndex);
+    currentIndex -= 1;
+
+    // And swap it with the current element.
+    temporaryValue = newPile[currentIndex];
+    newPile[currentIndex] = newPile[randomIndex];
+    newPile[randomIndex] = temporaryValue;
+  }
+  return newPile;
+}
```

Comment:
What about something simpler like:

```
export function shuffleArray<T = unknown>(cards: readonly T[]): T[] {
  return cards.slice().sort(() => Math.random() - 0.5);
}
```

Reply:
Indeed it's a good idea to simplify it
I don't know if the probabilities that a card stays at the same place are acceptable

Let's go with that that indeed simplifies a lot and we will what's going to happen later